### PR TITLE
Read network from external files

### LINKED ISSUE #506: NES and FEP+ network ingestion
We need light wrappers to take in serialized NES and FEP+ networks and spit out LigandNetworks

### PR DESCRIPTION
**Pull request by IAlibay**
Resolves #506

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

File: openfe/setup/ligand_network_planning.py [hunk 1]
```diff
@@ -303,3 +304,86 @@ def generate_network_from_indices(
...
+      If an unexpected line format is encountered.
+    """
+
+    with open(network_file, 'r') as f:
+        network_lines = [l.split() for l in f.readlines()]
+
+    names = []
+    for entry in network_lines:
+        if len(entry) != 5 or entry[1] != '#' or entry[3] != '->':
+            errmsg = ("line does not match expected name >> name format: "
```
Comment (hannahbaumann) [inline openfe/setup/ligand_network_planning.py:383]: maybe change errormsg as in the fep+ edge file the format is name -> name?

File: openfe/setup/ligand_network_planning.py
Comment (hannahbaumann) [inline openfe/setup/ligand_network_planning.py:1]: Just a general thought on naming NES: Once we implement an openfe NES protocol, referring to the Orion NES related things as just "nes" might be confusing...but maybe also not, just a thought to keep in mind.

Comment (IAlibay) [reply openfe/setup/ligand_network_planning.py:1]: That's a good point. how about just plainly `load_orion_network`?

File: openfe/setup/ligand_network_planning.py (hunk 1 above)
Comment (IAlibay) [reply openfe/setup/ligand_network_planning.py:383]: Oops. I looks like I copy-pasted and forgot to change 😅

File: openfe/setup/ligand_network_planning.py
Comment (hannahbaumann) [reply openfe/setup/ligand_network_planning.py:1]: Yes, that sounds good!

File: openfe/tests/setup/test_network_planning.py [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # This code is part of OpenFE and is licensed under the MIT license.
 # For details, see https://github.com/OpenFreeEnergy/openfe
+import io
+import tempfile
```
Comment (dwhswenson) [inline openfe/tests/setup/test_network_planning.py:4]:
```suggestion
```

looks like this import isn't used?

**Review by dwhswenson — CHANGES_REQUESTED** [review]
Could you add tests for the case that a ligand name in the input file is not a known ligand? (Should be handled by an error in `generate_network_from_names`, but it would be useful to have a test for that functionality.)

File: openfe/tests/setup/test_network_planning.py [hunk 3]
```diff
@@ -318,3 +318,104 @@ def test_network_from_indices_indexerror(atom_mapping_basic_test_files):
...
+    assert actual_edges == expected_edges
+
+
+@pytest.mark.parametrize('file_fixture, loader', [
+    ['orion_network',
+     openfe.setup.ligand_network_planning.load_orion_network],
+    ['fepplus_network',
+     openfe.setup.ligand_network_planning.load_fepplus_network],
+])
+def test_network_from_external_unknown_edge(file_fixture, loader, request,
```
Comment (IAlibay) [inline openfe/tests/setup/test_network_planning.py:362]: Here's the extra test.

**Review by dwhswenson — APPROVED** [review]
lgtm

**Pull request merged**